src/System.Text.Json/src/System/Text/Json/Serialization/ReadStackFrame.cs

```diff
@@ -241,16 +231,6 @@ public static object CreateDictionaryValue(ref ReadStack state)
         {
             JsonPropertyInfo jsonPropertyInfo = state.Current.JsonPropertyInfo;
 
-            if (jsonPropertyInfo == null)
```

Comment:
Don't need this check
- guards are implemented within the code flow for nested dictionaries in `HandleStartDictionary`
- for root dictionaries, we wouldn't be in this method if either condition were `false`